### PR TITLE
Implementation of st_union and st_union_agg

### PR DESCRIPTION
**Pull request by tdikland**
This PR contains the implementation some union variants.
- binary union: `st_union` for calculating the point set union of two geometries
- aggregate union: `st_union_agg` for calculating the point set union of a set of rows

### CODE REVIEW

File: src/test/scala/com/databricks/labs/mosaic/expressions/geometry/ST_UnionTest.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    private val codegenOnly =
+        withSQLConf(
+          SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "false",
+          SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "true",
+          SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.CODEGEN_ONLY.toString
+        ) _
+
+    test("Testing stUnion (H3, JTS) NO_CODEGEN") { noCodegen { unionBehavior(H3IndexSystem, JTS) } }
+    test("Testing stUnion (H3, ESRI) NO_CODEGEN") { noCodegen { unionBehavior(H3IndexSystem, ESRI) } }
+    test("Testing stUnion (BNG, JTS) NO_CODEGEN") { noCodegen { unionBehavior(BNGIndexSystem, JTS) } }
```
Comment (edurdevic) [inline src/test/scala/com/databricks/labs/mosaic/expressions/geometry/ST_UnionTest.scala:27]: No need to test this over multiple grid systems.
Please remove all BNG tests, and test only for H3. Or even better: create a new index system (under tests, not src) called `MockIndexSystem` that implements the trait and throws unimplemented exception on all methods. 
This can then be used for all the ST_ functions tests (better to do this in a separate PR).

File: src/main/scala/com/databricks/labs/mosaic/core/geometry/MosaicGeometryJTS.scala [hunk 2]
```diff
@@ -106,6 +106,11 @@ abstract class MosaicGeometryJTS(geom: Geometry) extends MosaicGeometry {
 
     override def equals(other: java.lang.Object): Boolean = false
 
+    override def equalsTopo(other: MosaicGeometry): Boolean = {
```
Comment (edurdevic) [inline src/main/scala/com/databricks/labs/mosaic/core/geometry/MosaicGeometryJTS.scala:109]: Well done! :)

**Review by edurdevic — APPROVED** [review]
LGTM!
Few test suggestions.

File: src/main/scala/com/databricks/labs/mosaic/expressions/geometry/ST_IntersectionAggregate.scala [hunk 3]
```diff
@@ -5,8 +5,8 @@ import com.databricks.labs.mosaic.core.index.{IndexSystem, IndexSystemID}
 import com.databricks.labs.mosaic.expressions.index.IndexGeometry
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.{Expression, ExpressionInfo}
-import org.apache.spark.sql.catalyst.expressions.aggregate.{ImperativeAggregate, TypedImperativeAggregate}
-import org.apache.spark.sql.catalyst.trees.BinaryLike
+import org.apache.spark.sql.catalyst.expressions.aggregate.{ImperativeAggregate, TypedImperativeAggregate, Sum, CollectList, Average}
```
[the remark this answers is not in the document]
Comment (tdikland) [reply src/main/scala/com/databricks/labs/mosaic/expressions/geometry/ST_IntersectionAggregate.scala]: Sorry, I accidentally committed this file. Removed it from this PR.

File: src/test/scala/com/databricks/labs/mosaic/expressions/geometry/ST_UnionTest.scala (hunk 1 above)
Comment (tdikland) [reply src/test/scala/com/databricks/labs/mosaic/expressions/geometry/ST_UnionTest.scala:27]: Great idea! Lots of tests are suffering from this flaw. Will take a stab at this later this week.
